### ISSUE TITLE
Asking about the return_scores during generation

### ISSUE THREAD

**Issue by freyaya123** [OpenNMT/CTranslate2#1661]
Hi, I'm new to ctranslate2, and I'm confused about the scores returned by generator.generate_batch() function. What's the coresponding meaning of the scores in the huggingface generate() function?

For example, in hf generation:

`import torch
from transformers import AutoModelForCausalLM
from transformers import AutoTokenizer

gpt2 = AutoModelForCausalLM.from_pretrained("gpt2", return_dict_in_generate=True)
tokenizer = AutoTokenizer.from_pretrained("gpt2")

input_ids = tokenizer("Today is a nice day", return_tensors="pt").input_ids  # [1, 5]

generated_outputs = gpt2.generate(input_ids, do_sample=True, num_return_sequences=3, output_scores=True)  # scores: 15*[3, vocab_size]
`

But if I use ctranslate2, for example:
`step_results = generator.generate_batch([prompt_tokens],num_hypotheses=3...`
I will get a list of len 3 for step_results[0].scores

And I also noticed that there is another function in hf:
`transition_scores = model.compute_transition_scores(generated_outputs.sequences,generated_outputs.scores,normalize_logits=True) #[3,15]`

What's the corresponding relationship between `generated_outputs.scores`, `transition_scores` and `step_results[0].scores`?

**Comment by minhthuc2502** [comment]
Hello,
1/ It seems like HF return the score of all vocab in each step. Otherwise, Ctranslate2 calculate the sum of the highest score of each step. 3 in your case is the batch size.
2/ Could you set the ``include_eos_in_hypotheses`` to True ? The eos token should be added at the end.

**Comment by freyaya123** [comment]
> Hello, 1/ It seems like HF return the score of all vocab in each step. Otherwise, Ctranslate2 calculate the sum of the highest score of each step. 3 in your case is the batch size. 2/ Could you set the `include_eos_in_hypotheses` to True ? The eos token should be added at the end.

What do you mean by "Ctranslate2 calculate the sum of the highest score of each step"?
for example,  if we assume bs=1
HF score:  seq_len*[1,vocab]
ctranslate2 score: a list of len 1. [Num]

What is the Num equal to?

**Comment by minhthuc2502** [comment]
For example bs = 1. HF score: seq_len x 1 x vocab. Otherwise, Ctranslate2 have shape: 1:  (max score in vocab) of token 1 +  (max score in vocab) of token 2 + ... +  (max score in vocab) of token seq_len

**Comment by freyaya123** [comment]
> For example bs = 1. HF score: seq_len x 1 x vocab. Otherwise, Ctranslate2 have shape: 1: (max score in vocab) of token 1 + (max score in vocab) of token 2 + ... + (max score in vocab) of token seq_len . If you want to get the max score for each token. You can use the async function and then get score of each token.

Thank you! Another question, according to the autoregressive score after linear layer and chain rule, why is ``sum`` here rather than ``product``? --P(x1)*P(x2|x1)*P(x3|x1,x2)*...*P(xn|x1,x2,...x_n-1)=P(x1,x2,...xn), if we want to calculate the score of generated sequence. I remember there is no ``log`` operation in the returned HF score.

**Comment by minhthuc2502** [comment]
In ctranslate2, the score after each step is the log-likelihood score. That's why we do the sum.

**Comment by freyaya123** [comment]
Oh I see! Thanks so much!

**Comment by freyaya123** [comment]
> include_eos_in_hypotheses

Sorry I don't find the parameter `include_eos_in_hypotheses` in the generate_batch function, where to set this?
